fe/fe-core/src/main/java/com/starrocks/leader/ReportHandler.java

```diff
@@ -1216,6 +1225,69 @@ private static void handleSetTabletEnablePersistentIndex(long backendId, Map<Lon
         }
     }
 
+    private static void handleSetTabletBinlogConfig(long backendId, Map<Long, TTablet> backendTablets) {
+        List<Triple<Long, Integer, BinlogConfig>> tabletToBinlogConfig = Lists.newArrayList();
+
+        TabletInvertedIndex invertedIndex = GlobalStateMgr.getCurrentInvertedIndex();
+        for (TTablet backendTablet : backendTablets.values()) {
+            for (TTabletInfo tabletInfo : backendTablet.tablet_infos) {
+                if (!tabletInfo.isSetBinlog_config_version()) {
+                    continue;
+                }
+                long tabletId = tabletInfo.getTablet_id();
+                long beBinlogConfigVersion = tabletInfo.binlog_config_version;
+                TabletMeta tabletMeta = invertedIndex.getTabletMeta(tabletId);
+                long dbId = tabletMeta != null ? tabletMeta.getDbId() : TabletInvertedIndex.NOT_EXIST_VALUE;
+                long tableId = tabletMeta != null ? tabletMeta.getTableId() : TabletInvertedIndex.NOT_EXIST_VALUE;
+
+                Database db = GlobalStateMgr.getCurrentState().getDb(dbId);
+                if (db == null) {
+                    continue;
+                }
+                db.writeLock();
+                try {
+                    OlapTable olapTable = (OlapTable) db.getTable(tableId);
+                    if (olapTable == null) {
+                        continue;
+                    }
+
+                    BinlogConfig binlogConfig = olapTable.getCurBinlogConfig();
+                    // backward compatible
+                    if (binlogConfig == null) {
+                        continue;
+                    }
+                    Long feBinlogConfigVersion = binlogConfig.getVersion();
+                    if (beBinlogConfigVersion < feBinlogConfigVersion) {
+                        tabletToBinlogConfig.add(new ImmutableTriple<>(tabletId, tabletInfo.schema_hash,
+                                olapTable.getCurBinlogConfig()));
+                    } else if (beBinlogConfigVersion == feBinlogConfigVersion) {
+                        if (olapTable.enableBinlog() && olapTable.getBinlogAvailableVersin().size() == 0) {
+                            long totalReplicaCount = olapTable.getAllPartitions().stream().
+                                    map(partition -> partition.getBaseIndex().getReplicaCount()).
+                                    reduce(0L, (acc, n) -> acc + n);
+                            GlobalStateMgr.getCurrentState().getBinlogManager().checkAndSetBinlogAvailableVersion(db,
+                                    olapTable, tabletId, backendId, totalReplicaCount);
+                        }
+                    } else {
+                        LOG.warn(String.format("table %s binlog config version of tabletId: %d, BeId: %d, is %d " +
```

Comment:
will this make binlog error?in which case will appear?

Reply:
It should not appear normally, just for debugging